src/global_distrib/mod_global_distrib_outgoing_conns_sup.erl

```diff
@@ -26,19 +26,42 @@ add_server(Server) ->
       type => supervisor,
       modules => dynamic
      },
+    ?INFO_MSG("event=outgoing_conn_start_progress,server='~s',state='starting'", [Server]),
     case supervisor:start_child(?MODULE, ServerSupSpec) of
-        {ok, _} -> ok;
-        {error, {already_started, _}} -> ok;
-        Error -> Error
+        {ok, Pid} ->
+            ?INFO_MSG("event=outgoing_conn_start_progress,server='~s',"
+                      "state='started',pid='~p'", [Server, Pid]),
+            ok;
+        {error, {already_started, Pid}} ->
+            ?INFO_MSG("event=outgoing_conn_start_progress,server='~s',"
+                      "state='started_by_other',pid='~p'", [Server, Pid]),
+            ok;
+        Error ->
+            ?INFO_MSG("event=outgoing_conn_start_progress,server='~s',"
+                      "state='failed',error='~p'", [Server, Error]),
+            Error
     end.
 
 -spec get_connection(Server :: jid:lserver()) -> pid().
 get_connection(Server) ->
```

Comment:
Can be still relevant
https://github.com/esl/MongooseIM/pull/1660/files#r165371814

I am feeling that we would find more bugs in this part of GD.

Reply:
Since it was mentioned in another PR, maybe it should rather be resolved there? But +1 for adding some comment about the behaviour.